pyproject.toml

```diff
@@ -22,9 +22,7 @@ dependencies = ["coloredlogs ~= 15.0", "pyyaml ~= 6.0"]
 dev = [
     "build ~= 0.8",
     "ipython ~= 8.4",
-    "mypy-extensions ~= 1.0",
```

Comment:
Not sure why this was here? not to do a chesterton's fence violation but I think it can be removed regardless of whether we keep pyright

Reply:
load-bearing unimported dependency

Reply:
probably was used for IDE hinting for anyone who had mypy installed